src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerInitializationHostService.cs

```diff
@@ -18,12 +18,12 @@ namespace Microsoft.Azure.WebJobs.Script.WebHost.ContainerManagement
     public class LinuxContainerInitializationHostService : IHostedService
     {
         private readonly IEnvironment _environment;
-        private readonly IInstanceManager _instanceManager;
+        private readonly LinuxInstanceManager _instanceManager;
         private readonly ILogger _logger;
         private readonly StartupContextProvider _startupContextProvider;
         private CancellationToken _cancellationToken;
 
-        public LinuxContainerInitializationHostService(IEnvironment environment, IInstanceManager instanceManager, ILogger<LinuxContainerInitializationHostService> logger, StartupContextProvider startupContextProvider)
+        public LinuxContainerInitializationHostService(IEnvironment environment, LinuxInstanceManager instanceManager, ILogger<LinuxContainerInitializationHostService> logger, StartupContextProvider startupContextProvider)
```

Comment:
Wouldn't the type for LegionInstanceManager and InstanceManager (will change to AtlasInstanceManager) be LinuxInstanceManager since both are subclasses of LinuxInstanceManager and not IInstanceManager?

Reply:
`LinuxInstanceManager` implements `IInstanceManager`, so through inheritance both `LegionInstanceManager` and `AtlasInstanceManager` are `IInstanceManager`.